hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaSingleThreadedMessageReceiver.java

```diff
@@ -106,6 +107,11 @@ public KafkaSingleThreadedMessageReceiver(KafkaConsumer<byte[], byte[]> consumer
                 }
             }
             return Optional.ofNullable(readQueue.poll());
+        } catch (InterruptException ex) {
+            logger.info("Kafka consumer thread interrupted", ex);
+            logger.info("Is current thread interrupted ? ---> {} ", Thread.currentThread().isInterrupted());
```

Comment:
Wow, despite the Thread.currentThread().interrupt() is called in ```InterruptException```'s constructor this log prints false.

Reply:
But when you do this in local test it always logs true... very weird but it is the solution to problem of hung processes